### PR TITLE
Vendor: context for services

### PR DESCRIPTION
**Pull request by chadwcarlson**
## Why

Addresses https://github.com/platformsh/platformsh-docs/issues/3365, but just for context in `/add-services`.

## What's changed

Applies configuration and vendorized context to `/add-services` content.

- [x] references to config-readers
- [x] Use in app > Vendor > `.environment` var exports
- [x] Using a framework intro
- [x] CLI substitutions
- [x] Variable substitutions
- [x] MySQL/MariaDB
- [x] `config_links`
- [x] MySQL/MariaDB/Oracle MySQL usage example differences
- [x] MongoDB enterprise
- [x] Persistent Redis

### CODE REVIEW

File: sites/platform/src/add-services/mysql/_index.md [hunk 1]
```diff
@@ -245,6 +342,7 @@ If neither `schemas` nor `endpoints` is included, it's equivalent to the followi
                 default_schema: main
                 privileges:
                     main: admin
+{{% /snippet %}}
```
Comment (AnouckColson) [inline sites/platform/src/add-services/mysql/_index.md:345]: Just realized that there's duplicate content here. **Define permissions** section should end with "You can also specify schemas for [multiple databases](link). The rest can be removed as it's identical to what's in the section directly below.

File: sites/platform/src/add-services/opensearch.md [hunk 2]
```diff
@@ -220,8 +263,8 @@ There are two ways to do so.
 
 ### Destructive
 
-In your `{{< vendor/configfile "services" >}}` file, change the version *and* name of your OpenSearch service.
-Then update the name in the `{{< vendor/configfile "app" >}}` relationships block.
+In your `{{< vendor/configfile "services" >}}` file, change the version *and* name of your Elasticsearch service.
```
Comment (AnouckColson) [inline sites/platform/src/add-services/opensearch.md:266]:
```suggestion
In your `{{< vendor/configfile "services" >}}` file, change the version *and* name of your Opensearch or Elasticsearch service.
```

Shouldn't we have both here? The intro of the **Upgrading** section implies that such upgrades can be needed when upgrading from one OS version to another, and not just from ES to OS?

**Review by AnouckColson — CHANGES_REQUESTED** [review]

File: themes/psh-docs/layouts/partials/examples/config_links.html [hunk 3]
```diff
@@ -39,7 +39,7 @@ <h3>MariaDB example configuration</h3>
   <h3>Ephemeral Redis example configuration</h3>
 {{ end }}
 <a href="/add-services.html">Service definition</a>
-
+config_links
```
Comment (chadwcarlson) [inline themes/psh-docs/layouts/partials/examples/config_links.html:42]:
```suggestion
```

File: sites/platform/src/add-services/_index.md [hunk 4]
```diff
@@ -95,16 +113,40 @@ This is done in your [app configuration for relationships](../create-apps/app-re
...
     postgresql_database: "database2:postgresql"
+{{% /snippet %}}
+{{% snippet name="database1" config="service" placeholder="true" %}}
+    type: mariadb:{{% latest "mariadb" %}}
+    disk: 2048
+{{% /snippet %}}
+{{% snippet name="database2" config="service" globKey="false" placeholder="true" %}}
+    type: postgresql:{{% latest "postgresql" %}}
+    disk: 1024
+{{% /snippet %}}
```
[the remark this answers is not in the document]
Comment (AnouckColson) [reply sites/platform/src/add-services/_index.md:149]: @chadwcarlson I asked Rémi this morning and he said we can keep premium services (although it's not know how they're gonna be billed, but he said there's no point removing them to put them back in very soon after).

File: sites/platform/src/add-services/opensearch.md (hunk 2 above)
Comment (chadwcarlson) [reply sites/platform/src/add-services/opensearch.md:266]: I think introducing Elasticsearch here was a copy-paste mistake on my part. Fixing.

**Pull request merged**